packages/buckets/src/buckets.ts

```diff
@@ -253,6 +254,46 @@ export class Buckets extends BucketsGrpcClient {
     return bucketsListPath(this, key, path)
   }
 
+  /**
+   * Returns information about a bucket path.
+   * @param key Unique (IPNS compatible) identifier key for a bucket.
+   * @param path A file/object (sub)-path within a bucket.
+   */
+  async listPathRecursive(key: string, path: string): Promise<ListPathRecursive> {
+    logger.debug('list path recursive request')
+    return await utilListPathRecursive(this, key, path)
+  }
+
+  /**
+   * listPathRecursive returns a nested object of all paths (and info) in a bucket
+   * @param key Unique (IPNS compatible) identifier key for a bucket.
+   * @param path A file/object (sub)-path within a bucket.
+   * @param dirs (optional) if false will include only file paths
+   *
+   * @example
+   * ```typescript
+   * import { Buckets } from '@textile/hub'
+   *
+   * async function printPaths(buckets: Buckets, bucketKey: string) {
+   *   const list = await buckets.listPathRecursiveFlat(bucketKey, '')
+   *   console.log(list)
+   * }
+   * // [
+   * //   'mybuck',
+   * //   'mybuck/.textileseed',
+   * //   'mybuck/dir1',
+   * //   'mybuck/dir1/file1.jpg',
+   * //   'mybuck/path',
+   * //   'mybuck/path/to',
+   * //   'mybuck/path/to/file2.jpg'
+   * // ]
+   * ```
+   */
+  async listPathRecursiveFlat(key: string, path: string, dirs = true): Promise<ListPathRecursiveFlat> {
```

Comment:
This is nice! But why not just `listPath(key: string, path: string, opts?: { dirs: boolean, recursive: boolean, flatten: boolean })`

Reply:
i just wasn't sold on the various return types.  i can definitely get two of them into one, normal and recursive. but the flat method is a bit different, what do you think?

Reply:
hmm... i also don't want to make the `dirs` option an option on the `listPath` method. so there is a bit of a mismatch there ("i used list path with, opts: {dirs: false}, but i'm still getting dirs?" type questions pretty quick)

Reply:
I'm sold